Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -295,13 +295,17 @@ WarpXParticleContainer::AddNParticles (int /*lev*/, long n,
         );
     }
 
+    // Move particles to their appropriate tiles
+    Redistribute();
+
     // Remove particles that are inside the embedded boundaries
 #ifdef AMREX_USE_EB
     auto & distance_to_eb = WarpX::GetInstance().GetDistanceToEB();
     scrapeParticles( *this, amrex::GetVecOfConstPtrs(distance_to_eb), ParticleBoundaryProcess::Absorb());
 #endif
 
-    Redistribute();
+    // Call (local) redistribute again to remove particles with invalid ids
+    Redistribute(lev, lev, 1, true, true);
```

Comment:
I still think that the `local` argument should be 1, since it is expecting an `int`. The description for local says "If > 0, this is the maximum number of cells a particle can have moved since the last Redistribute() call." It is not a flag.
I was thinking that if nGrow=1 and local=1, the routine might be smart enough to know that no communication is needed. No communication is needed since it is only removing invalid particles. Is there maybe  a separate routine that does just that? This would need a comment from @WeiqunZhang

Reply:
Oh, I see. I definitely interpreted `local` incorrectly. I guess it should actually be `local=0` then, since the particles will have moved at all since the previous time `Redistribute()` was called.

There was some mention recently of creating a function that will only remove invalid particles, but as far as I know that has not been done yet.
```suggestion
    Redistribute(lev, lev, 1, 0, true);
```

Reply:
This confuses me though since the default value for `local` is `0`, meaning usually when we call `Redistribute` it is as if we specify that the particles have not moved.

Reply:
Local=0 means a global communication will be done as if particles may have moved across multiple processors. I think the correct value is 1.

Reply:
I see. But then looking at the actual redistribute function code, it has [this](https://github.com/AMReX-Codes/amrex/blob/2cd72a3bbaeb9e44475225e4acb02b35084e19da/Src/Particle/AMReX_ParticleContainerI.H#L1433):
```
    if (local > 0) { BuildRedistributeMask(0, local); }
```
While I don't fully understand what exactly is done with `BuildRedistributeMask`, it seems to me that it determines the neighboring ranks to which particles could be passed / received from.
Is it correct then to say that even with `local=1` there would still be some unnecessary overhead but that overhead will be minimized compared to the case where particles could be globally redistributed?

This really makes a case for having a function that simply, locally, removes invalid particles.